soh/soh/Enhancements/randomizer/randomizer_check_objects.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+#include "randomizerTypes.h"
+#include <string>
+#include <map>
+#include <unordered_map>
+
+// Check types based on main settings
+typedef enum {
+    RCTYPE_STANDARD, // Base set of rando checks
+    RCTYPE_SKULL_TOKEN, // Gold Skulltulas
+    RCTYPE_COW, // Cows
+    RCTYPE_ADULT_TRADE, // Adult trade quest checks
+    RCTYPE_FROG_SONG, // Frog song purple rupee checks
+    RCTYPE_MAP_COMPASS, // Maps/Compasses
+    RCTYPE_SMALL_KEY, // Small Keys
+    RCTYPE_GF_KEY, // Gerudo Fortress Keys
+    RCTYPE_BOSS_KEY, // Boss Keys
+    RCTYPE_GANON_BOSS_KEY, // Ganon's boss key
+    RCTYPE_SHOP, // shops/scrubs
+    RCTYPE_CHEST_GAME, //todo replace this once we implement it, just using it to exclude for now
+    RCTYPE_LINKS_POCKET, //todo this feels hacky
+    RCTYPE_GOSSIP_STONE,
+} RandomizerCheckType;
```

Comment:
For clarity, this is the type of the original item at this location, or the type of the randomized item? Not sure it makes much sense to have keys/maps/compasses in here, as they are still randomized within their dungeons without keysanity right? Or I guess the idea is that you can exclude any given key chest and it will be a vanilla key?

Reply:
i was just doing keys/maps/compasses based on the existing rando settings keys we had (so i can exclude maps/compasses from the list if they're set as vanilla)

Reply:
Oh are there settings to keep all the keys/maps/compasses vanilla? There probably are I just forgot. They all make perfect sense then.